packages/mermaid/src/diagrams/treeView/detector.ts

```diff
@@ -0,0 +1,21 @@
+import type { DiagramDetector, DiagramLoader } from '../../diagram-api/types.js';
+import type { ExternalDiagramDefinition } from '../../diagram-api/types.js';
+
+const id = 'treeView';
+
+const detector: DiagramDetector = (txt) => {
+  return /^\s*treeView/.test(txt);
```

Comment:
All the new diagrams should have `-beta` suffix, so changes to syntax can be added later. 

```suggestion
  return /^\s*treeView-beta/.test(txt);
```